pkg/controllers/management/auth/role_template_handler.go

```diff
@@ -0,0 +1,65 @@
+package auth
+
+import (
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/cache"
+)
+
+const (
+	roleTemplateController  = "mgmt-auth-role-template-controller"
+	prtbByRoleTemplateIndex = "management.cattle.io/prtb-by-role-template"
+)
+
+type roleTemplateLifecycle struct {
```

Comment:
dont call it a life cycle if its just a sync handler. lifecycles have Create Updated and Remove functions and have a lot of implications (kubeneretes finalizers), so its important that the name be accurate.